app/src/main/java/io/appium/settings/Settings.java

```diff
@@ -70,12 +82,98 @@ public void onCreate(Bundle savedInstanceState) {
             LocationTracker.getInstance().start(this);
         }
 
+        handleRecording(getIntent());
+    }
+
+    private void handleRecording(Intent openingIntent) {
+        if (openingIntent != null && openingIntent.getAction() != null) {
```

Comment:
can we use guard conditions to make this method more readable?

Reply:
done